controller/src/main/java/org/jboss/as/controller/extension/ExtensionRegistry.java

```
@@ -431,18 +436,20 @@ public SubsystemRegistration registerSubsystem(String name, int majorVersion, in
 
         @Override
         public SubsystemRegistration registerSubsystem(String name, int majorVersion, int minorVersion, int microVersion, boolean deprecated) {
+            return registerSubsystem(name, ModelVersion.create(majorVersion, minorVersion, microVersion), deprecated);
+        }
+
+        public SubsystemRegistration registerSubsystem(String name, ModelVersion version, boolean deprecated) {
             assert name != null : "name is null";
             checkNewSubystem(extension.extensionModuleName, name);
             SubsystemInformationImpl info = extension.getSubsystemInfo(name);
-            info.setMajorVersion(majorVersion);
-            info.setMinorVersion(minorVersion);
-            info.setMicroVersion(microVersion);
+            info.setVersion(version);
```

I think I would throw an exception if version is null, and get rid of the null checks in SubsystemInformationImpl